response_operations_ui/controllers/uaa_controller.py

```diff
@@ -179,9 +190,70 @@ def change_user_password(email, password):
     return change_password(access_token=access_token, user_code=password_reset_code, new_password=password)
 
 
-def create_user_account(email, password, user_name, first_name, last_name):
+def reset_user_password_by_id(user_id: str, password: str) -> requests.Response | None:
```

Comment:
I changed the names so they're more consistent but I'm really happy with that suggested code change as I don't think it's much better then what we have here.  Would you be okay with just the name change or do you feel strongly that we should change this bit?

Reply:
It's only an opinion and this does works, so I have no issue if you don't want to change it. Absorbing retrieve_user_code (password_reset_code) into change password does make sense to me, because you always need to do it to change a password and if we had 5+ places that called change password would we want to call and pass it 5 times. Up to you, but won't block